Makefile

```diff
@@ -55,7 +55,7 @@ clean-apisonator:
 
 auth: ## Build threescale_wasm_auth filter. 
 	@echo "> Building threescale_wasm_auth filter"
-	git submodule foreach git pull origin main
+	git submodule update --init
```

Comment:
We need to pull to clarify that we are in sync with the auth upstream. If we are behind, then the CI will fail saying unrelated git history so we know that we are behind. Otherwise we'll be running the CI with an outdated version of the auth.

Reply:
I am aware of the fact that wasm-auth module can be behind compared to upstream since I added the `foreach...` statement because of the same exact issue. But if you look at commit `e25e97b`, you'll see that it doesn't know how to manage conflict and it can create a merge commit if we force it to pull. That's why I think it's best to manually update the submodule to avoid critical changes by the code, on which we have no control.

Reply:
we shouldn't force(by default its not forcing unless we pass `--force`). it won't merge if it sees unrelated git histories. the build will fail and then we know we are behind. then we can update manually. otherwise we won't know.

Reply:
Build won't fail but actually, the integration job may fail which can be tested using `make integration` before making a PR.

Reply:
No in this case build fails. For building the auth git submodule update is required and it will fail if it sees the unrelated git history. Your CI for this PR failed because of that. 
https://github.com/3scale-labs/gsoc-wasm-filters/actions/runs/1056006285

Both ways the sub module update is done manually. Only difference is if we pull we know that the sub module is lagging behind. If we don't pull then someone has to always look whether the auth is updated manually which is not very maintainable.

Reply:
You'll only know that submodule is lagging behind only when there is an unrelated git history else it will pull and use the latest version even though you have a previous commit for the sub-module in this repo. So, someone has to be always aware of that lag.

Reply:
If someone updates the auth upstream and we have a previous version, then we'll always get a unrelated history.

Reply:
Updated.